.gitignore

```diff
@@ -32,6 +32,13 @@ Project/.idea/**/libraries
 # .idea/*.iml
 # .idea/modules
 
```

Comment:
Nothing wrong with this particular file/lines but general notes:

a) You have two issues, bare minimum of two PRs for next time please
b) You could have split this pr into 4/5 easily. I will state where you could have throughout the review
c) Refactoring other pieces of code should be done in a separate PR
d) If you are going to change the design of things that have already been built, please present the design to your teammates first so we can accept/deny/change it together. If you do not do that, and the design is flawed, all your code and time is wasted
e) Since design choice was not presented, some guiding comments would have been nice. In 400+ lines I counted maybe 2/3 comments. Please document all methods as well.
f) Try to limit your work to only your work. If you have issues 6/8 assigned, please only work on issues 6/8. This is a group project and everyone should be allowed to do their work.
g) If at any moment, what you are coding goes outside the scope of your issue, but is needed - Make a new issue, assign it to yourself/nobody and all changes wrt to that issue will be in a new PR.